front/src/applications/stdcm/hooks/useLinkedPathSearch.ts

```diff
@@ -0,0 +1,163 @@
+      if (!('operational_point' in origin) || !('operational_point' in destination))
+        return undefined;
+      const originId = origin.operational_point;
+      const destinationId = destination.operational_point;
+
+      try {
+        const payloadOP = {
+          object: 'operationalpoint',
+          query: ['or', ['=', ['obj_id'], originId], ['=', ['obj_id'], destinationId]],
+        };
+        const resultsOP = await postSearch({ searchPayload: payloadOP, pageSize: 25 }).unwrap();
+        const groupedResults = groupBy(resultsOP, 'obj_id');
+        return {
+          origin: groupedResults[originId][0],
+          destination: groupedResults[destinationId][0],
+        };
+      } catch (error) {
+        console.error('Failed to fetch operational points:', error);
+        return undefined;
+      }
+    },
+    [postSearch]
+  );
+
+  const launchTrainScheduleSearch = useCallback(async () => {
+    setDisplaySearchButton(false);
+    setLinkedPathResults([]);
+    try {
+      const results = (await postSearch({
+        searchPayload: {
+          object: 'trainschedule',
+          query: [
+            'and',
+            ['search', ['train_name'], trainNameInput],
+            ['=', ['timetable_id'], timetableId!],
+          ],
+        },
+        pageSize: 25,
+      }).unwrap()) as SearchResultItemTrainSchedule[];
+      const filteredResults = results.filter((result) =>
+        isEqualDate(linkedPathDate, new Date(result.start_time))
+      );
+
+      if (!filteredResults.length) {
+        setDisplaySearchButton(true);
+        return;
+      }
+
+      const newLinkedPathResults = await Promise.all(
+        filteredResults.map(async (result) => {
+          const opDetails = await getExtremitiesDetails(result.path);
+          const computedOpSchedules = computeOpSchedules(
+            result.start_time,
+            result.schedule.at(-1)!.arrival!
+          );
+          if (opDetails === undefined) return undefined;
+          return {
+            trainName: result.train_name,
+            origin: { ...opDetails.origin, ...computedOpSchedules.origin } as StdcmLinkedPathStep,
+            destination: {
+              ...opDetails.destination,
+              ...computedOpSchedules.destination,
+            } as StdcmLinkedPathStep,
+          };
+        })
+      );
+      setLinkedPathResults(compact(newLinkedPathResults));
+      setHasSearchBeenLaunched(true);
+    } catch (error) {
+      console.error('Train schedule search failed:', error);
+      setDisplaySearchButton(true);
+    }
+  }, [postSearch, trainNameInput, timetableId, linkedPathDate, getExtremitiesDetails]);
+
+  const updateLinkedPathStep = useCallback(
+    (linkedPathStep: ExtremityPathStepType, trainName: string, pathStep: StdcmLinkedPathStep) => {
+      const { name, ch, uic, geographic, isoArrivalTime, date, time, trigram } = pathStep;
+      const newPathStep = {
+        name,
+        ch,
+        id: nextId(),
```

Comment:
You will need to give the correct id here now, because of my PR :/ 
You should get the stdcmOrigin and stdcmDestination at the top of the file, and then:
```suggestion
        id: linkedPathStep === 'origin' ? origin.id : destination.id,
```
You may need to add more props